### PR TITLE
Add link to download ES profiler data

### PR DESCRIPTION
**Pull request by esoergel**
## Product Description

This builds off the case search profiler util introduced in  https://github.com/dimagi/commcare-hq/pull/34373

It does a few things:
* Numbers ES queries so you can cross reference between the timing data and the query details
* Adds query duration in-line with query details
* Adds a link to download the profiler output for each query

This screenshot shows all of that:
image

## Safety Assurance
All of the functionality changes except the `_query_number` counter are gated behind a `debug_mode` flag, and so should only affect usages of this utility, not general case search

### Rollback instructions
- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/case_search/templates/case_search/profile_case_search.html [hunk 1]
```diff
@@ -66,9 +66,14 @@ <h3>Profile Slow Case Searches</h3>
...
         <div class="card-header clickable" data-bs-toggle="collapse" data-parent="#queries-accordion"
           data-bind="attr: { href: '#query-' + $index() }">
-          Elasticsearch query: <code data-bind="text: slug"></code>
+          Elasticsearch query #<span data-bind="text: query_number"></span>:
+          <code data-bind="text: slug"></code> -
+          <strong data-bind="text: duration ? duration.toFixed(3) : '-'"></strong> seconds
         </div>
         <div class="card-body panel-collapse collapse" data-bind="attr: { id: 'query-' + $index() }">
+          <!-- ko if: profile_url -->
+            <a class="btn btn-info" data-bind="attr: {href: profile_url}">Download Profile JSON</a>
```
Comment (orangejenny) [inline corehq/apps/case_search/templates/case_search/profile_case_search.html:75]:
```suggestion
            <a class="btn btn-info float-end" data-bind="attr: {href: profile_url}">Download Profile JSON</a>
```

File: corehq/apps/case_search/utils.py [hunk 2]
```diff
@@ -57,24 +60,47 @@ class CaseSearchProfiler:
...
+            })
+
+    @staticmethod
+    def _get_profile_url(slug, query_number, profile_json):
+        timestamp = get_timestamp_for_filename()
+        name = f'es_profile_{query_number}_{slug}_{timestamp}.json'
+        io = BytesIO()
+        io.write(json.dumps(profile_json).encode('utf-8'))
+        io.seek(0)
+        return get_download_url(io, name, content_type='application/json')
```
Comment (orangejenny) [inline corehq/apps/case_search/utils.py:103]: These files will get deleted after 24 hours, based on `timeout` in `get_download_url`, right?

**Review by orangejenny — APPROVED** [review]
Left an optional styling suggestion.

File: corehq/apps/case_search/utils.py (hunk 2 above)
Comment (esoergel) [reply corehq/apps/case_search/utils.py:103]: that's right

File: corehq/apps/case_search/templates/case_search/profile_case_search.html (hunk 1 above)
Comment (esoergel) [reply corehq/apps/case_search/templates/case_search/profile_case_search.html]: Thanks!

**Review by orangejenny — APPROVED** [review]

**Review by gherceg — APPROVED** [review]

**Review by MartinRiese — APPROVED** [review]

**Pull request merged**